packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EditInvoiceDetails.tsx

```diff
@@ -0,0 +1,289 @@
+    );
+  });
+
+  const methods = useForm<CreateReceivablesFormProps>({
+    resolver: yupResolver(getUpdateInvoiceValidationSchema(i18n)),
+    defaultValues: useMemo(
+      () => ({
+        /** Customer section */
+        counterpart_id: invoice.counterpart_id,
+        counterpart_contact: invoice.counterpart_contact,
+        counterpart_vat_id_id: invoice.counterpart_vat_id?.id ?? '',
+        default_shipping_address_id: counterpartShippingAddress?.id ?? '',
+        default_billing_address_id: '',
+
+        /** Entity section */
+        entity_vat_id_id: invoice.entity_vat_id?.id ?? '',
+        fulfillment_date: invoice.fulfillment_date
+          ? new Date(invoice.fulfillment_date)
+          : null,
+        purchase_order: invoice.purchase_order ?? '',
+
+        /** Items section */
+        line_items: invoice.line_items.map((lineItem) => ({
+          quantity: lineItem.quantity,
+          product_id: lineItem.product.id,
+          vat_rate_id: lineItem.product.vat_rate.id,
+          vat_rate_value: lineItem.product.vat_rate.value,
+          name: lineItem.product.name,
+          price: lineItem.product.price,
+          measure_unit_id: lineItem.product.measure_unit_id,
+        })),
+        vat_exemption_rationale: invoice.vat_exemption_rationale ?? '',
+
+        /** Items section */
+        entity_bank_account_id: invoice.entity_bank_account?.id ?? '',
+        payment_terms_id: invoice.payment_terms?.id ?? '',
+      }),
+      [
+        counterpartShippingAddress?.id,
+        invoice.counterpart_contact,
+        invoice.counterpart_id,
+        invoice.counterpart_vat_id?.id,
+        invoice.entity_bank_account?.id,
+        invoice.entity_vat_id?.id,
+        invoice.fulfillment_date,
+        invoice.line_items,
+        invoice.payment_terms?.id,
+        invoice.purchase_order,
+        invoice.vat_exemption_rationale,
+      ]
+    ),
+  });
+
+  const [actualCurrency, setActualCurrency] = useState(invoice.currency);
+
+  const {
+    handleSubmit,
+    setValue,
+    formState: { isDirty },
+  } = methods;
+
+  useEffect(() => {
+    setValue('default_shipping_address_id', counterpartShippingAddress?.id);
+  }, [counterpartShippingAddress?.id, setValue]);
+
+  const [isAlertOpen, setIsAlertOpen] = useState(false);
+  const handleCancelWithAlert = useCallback(() => {
+    /** If the form hasn't been changed, we could just cancel it */
+    if (!isDirty) {
+      onCancel();
+
+      return;
+    }
+
+    /** Otherwise open cancel alert modal */
+    setIsAlertOpen(true);
+  }, [isDirty, onCancel]);
+
+  const updateReceivableLineItems = useUpdateReceivableLineItems(invoice.id);
+  const updateReceivable = useUpdateReceivable(invoice.id);
+
+  const isLoading =
+    updateReceivableLineItems.isPending || updateReceivable.isPending;
+
+  // eslint-disable-next-line lingui/no-unlocalized-strings
+  const formName = 'Monite-Form-receivablesDetailsForm';
```

Comment:
Would be better to refactor as

```
const formName = `Monite-Form-receivablesDetailsForm-${useId()}`
```

Reply:
Updated https://github.com/team-monite/monite-sdk/commit/eab3d83aca9da2d41e5cc17430d92f03da2378f0